docs/src/custom.jl

```diff
@@ -0,0 +1,201 @@
+# * conversion to a (sparse) matrix `Matrix(A)` and `sparse(A)`.
+
+A = MyFillMap(5.0, (3, 3)); x = ones(3); sum(x)
+
+#-
+
+A * x
+
+#-
+
+mul!(zeros(3), A, x)
+
+#-
+
+mul!(ones(3), A, x, 2, 2)
+
+#-
+
+mul!(ones(3,3), A, reshape(collect(1:9), 3, 3), 2, 2)
+
+# ## Multiply-and-add and the `MulStyle` trait
+
+# While the above function calls work out of the box due to generic fallbacks, the latter
+# may be suboptimally implemented for your custom map type. Let's see some benchmarks.
+
+using BenchmarkTools
+
+@benchmark mul!($(zeros(3)), $A, $x)
+ 
+#-
+
+@benchmark mul!($(zeros(3)), $A, $x, $(rand()), $(rand()))
+
+# The second benchmark indicates the allocation of an intermediate vector `z`
+# which stores the result of `A*x` before it gets scaled and added to (the scaled)
+# `y = zeros(3)`. For that reason, it is beneficial to provide a custom "5-arg `mul!`"
+# if you can avoid the allocation of an intermediate vector. To indicate that there
+# exists an allocation-free implementation, you should set the `MulStyle` trait,
+# whose default is `ThreeArg()`.
+
+LinearMaps.MulStyle(A::MyFillMap) = FiveArg()
+
+function LinearAlgebra.mul!(
+    y::AbstractVecOrMat,
+    A::MyFillMap,
+    x::AbstractVector,
+    α::Number,
+    β::Number
+)
+    if iszero(α)
+        !isone(β) && rmul!(y, β)
+        return y
+    else
+        temp = A.λ * sum(x) * α
+        if iszero(β)
+            y .= temp
+        elseif isone(β)
+            y .+= temp
+        else
+            y .= y .* β .+ temp
+        end
+    end
+    return y
+end
+
+# With this function at hand, let's redo the benchmark.
+
+@benchmark mul!($(zeros(3)), $A, $x, $(rand()), $(rand()))
+
+# There you go, the allocation is gone and the computation time is significantly reduced.
+
+# ## Adjoints and transposes
+
+# The following functions are obviously helpful additions.
+
+LinearAlgebra.issymmetric(A::MyFillMap) = A.size[1] == A.size[2]
+LinearAlgebra.ishermitian(A::MyFillMap) = isreal(A) && A.size[1] == A.size[2]
+LinearAlgebra.isposdef(A::MyFillMap) = (size(A, 1) == size(A, 2) == 1 && isposdef(A.λ))
+Base.:(==)(A::MyFillMap, B::MyFillMap) = A.λ == B.λ && A.size == B.size
+
+# These are used, for instance, in checking symmetry or positive definiteness of
+# higher-order `LinearMap`s, like products or linear combinations of linear maps, or signal
+# to iterative eigenproblem solvers that real eigenvalues are to be computed.
+# Without these definitions, the first three functions would return `false` (by default),
+# and the last one would fall back to `===`.
+#
+# As for the multiplication of adjoints/transposes of `MyFillMap`s with vectors, there exist
+# two *distinct* paths.
+
+# ### Path 1: Generic, non-invariant `LinearMap` subtypes
+
+# Generically, taking the transpose (or the adjoint) of a (real, resp.) map wraps the
+# linear map by a `TransposeMap`.
+
+typeof(A')
+
+# Not surprisingly, without further definitions, multipliying `A'` by `x` yields an error.
+
+A'x
```

Comment:
unsure why this is not erroring...

Reply:
Thanks for pointing this out. It doesn't error because I have defined `issymmetric` for `MyFillMap`, and because it realizes that my specific `A` is symmetric, it doesn't require the formal transpose, but uses `A` itself! I must have rearranged things and used to have the `issymmetric` definitions below this line. That's instructive, I think it's going to error if I make it nonsquare, or complex.

Reply:
Somehow Documenter doesn't show the error. Let's see if one can do something about it: https://github.com/JuliaDocs/Documenter.jl/issues/1420
Otherwise, I'll just type the error message manually.